charty/src/commonMain/kotlin/com/himanshoe/charty/circle/SpeedometerProgressBar.kt

```diff
@@ -0,0 +1,179 @@
+package com.himanshoe.charty.circle
+
+import androidx.compose.animation.core.Animatable
+import androidx.compose.foundation.Canvas
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.padding
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.geometry.Offset
+import androidx.compose.ui.geometry.Size
+import androidx.compose.ui.graphics.Brush
+import androidx.compose.ui.graphics.StrokeCap
+import androidx.compose.ui.graphics.drawscope.DrawScope
+import androidx.compose.ui.graphics.drawscope.Stroke
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import com.himanshoe.charty.circle.config.DotConfig
+import com.himanshoe.charty.common.ChartColor
+import com.himanshoe.charty.common.TextConfig
+import kotlinx.coroutines.delay
+import kotlin.math.PI
+import kotlin.math.cos
+import kotlin.math.sin
+
+/**
+ * A composable function that displays a speedometer-style progress bar with an animated progress indicator.
+ *
+ * @param progress A lambda function that returns the current progress as a float value between 0 and 1.
+ * @param title The title text to be displayed at the center of the speedometer.
+ * @param color The color of the progress arc.
+ * @param progressIndicatorColor The color of the progress indicator circle.
+ * @param trackColor The color of the track arc.
+ * @param modifier The modifier to be applied to the Box containing the speedometer.
+ * @param dotConfig Configuration for the dots displayed along the arc.
+ * @param titleTextConfig Configuration for the title text style.
+ * @param subTitleTextConfig Configuration for the subtitle text style.
+ */
+@Composable
+fun SpeedometerProgressBar(
+    progress: () -> Float,
+    title: String,
+    color: ChartColor,
+    progressIndicatorColor: ChartColor,
+    trackColor: ChartColor,
+    modifier: Modifier = Modifier,
+    dotConfig: DotConfig = DotConfig.default(),
+    titleTextConfig: TextConfig = TextConfig.default(),
+    subTitleTextConfig: TextConfig = TextConfig.default(fontSize = 20.sp),
+) {
+    val progressAnimator = remember { Animatable(0f) }
+    LaunchedEffect(progress()) {
+        delay(400)
```

Comment:
Maybe you can make this configurable as well if consumer wants to animate it on their own? Maybe this can be a feature for future versions.

Reply:
Yes theese are future improvement, considering that I want to see how many use it